<!doctype html>
<html lang="en">

<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">

    <title>ParSe v2 - FASTA - REVIEWED PROTEOME</title>
    <meta name="description" content="">

    <link href="https://cdn.jsdelivr.net/npm/normalize.css@8.0.1/normalize.css" rel="stylesheet">
    <link href="https://unpkg.com/tabulator-tables@5.4.3/dist/css/tabulator.min.css" rel="stylesheet">
    <link rel="stylesheet" href="./css/app.css">
</head>

<body>
    <div id="loading" class="loading-overlay">
        <div id="loading-text" class="loading-text">Loading</div>
    </div>
    <main>
        <h1>ParSe v2: predict phase-separating protein regions from the primary sequence</h1>
        <p>REFERENCE PROTEOME IS THE SWISS-PROT REVIEWED SET OF HUMAN SEQUENCES (20,426 sequences, downloaded October 2023).</p>
        <p>To analyze a single protein sequence using ParSe v2, please visit <a href="https://stevewhitten.github.io/Parse_v2_web/">our other site</a>.</p>
        <p>The algorithm is described in Ibrahim et al, <a href="https://doi.org/10.1016/j.jbc.2022.102801"><i>J. Biol. Chem.</i> <b>299</b>, 102801 (2023)</a>.</p>
        <p>This web application and its output is described in Wilson et al, <a href="https://onlinelibrary.wiley.com/doi/10.1002/pro.4756"><i>Protein Science</i>, doi: 10.1002/pro.4756</a>.</p>
        <div class="mb-10">
            <h3>Input File</h3>
            <p>Input a <a href="https://en.wikipedia.org/wiki/FASTA_format">FASTA</a> file containing protein sequences, preferably in the <a href="https://www.uniprot.org/help/fasta-headers">UniProtKB</a> format. Files with tens of thousands of sequences may take a minute or longer to analyze. The algorithm ignores sequences longer than 10,000 residues or shorter than 25 residues.
            </p>
            <div class="mb-10">
                <label for="input-file">File: </label>
                <input id="input-file" type="file" accept=".txt, .fasta" />
            </div>
            <div class="mb-10">
                <label for="proteome-name">Name for the file (optional): </label>
                <br/>
                <input id="proteome-name" />
            </div>
            <div class="mb-10">
                <button id="submit">Submit</button>
            </div>
        </div>
        <h3>A</h3>
        <p class="mb-10" id="auc-a"></p>
        <div class="mb-10">
            <div class="chart-container">
                <div class="mb-10 chart" id="chart-a"></div>
                <div class="mb-10 chart" id="chart-auc-a"></div>
            </div>
            <p><b>A. Distribution of predicted PS IDR lengths.</b> ParSe v2 finds regions within proteins that are >90% labeled P, which are predicted to be PS IDRs. The y-axis is the percent of proteins in a set with PS IDRs at least as long as the length indicated by the x-axis. The recall plot compares the percent of set for the human proteome against itself and against the input file.</p>
            <h3>B</h3>
            <p class="mb-10" id="auc-b"></p>
            <div class="chart-container">
                <div class="mb-10 chart" id="chart-b"></div>
                <div class="mb-10 chart" id="chart-auc-b"></div>
            </div>
            <p><b>B. Distribution of PS potentials.</b> ParSe v2 calculates the classifier distance sum of windows labeled P, which we use as a numerical score to estimate the PS potential. The y-axis is the percent of proteins in a set with a PS potential equal to or greater than the value indicated by the x-axis. The recall plot compares the percent of set for the human proteome against itself and against the input file.</p>
            <div class="mb-10">
            <label for="download-summary-csv-b"><b>Sequence-calculated properties</b> (sort this table by its 3<sup>rd</sup> or 4<sup>th</sup> column to find phase-separating proteins): </label>
                <button id="download-summary-csv-b">Download CSV</button>
            </div>
            <div class="mb-10" id="summary-b"></div>
            <div class="mb-10">
            <label for="download-data-csv-a"><b>Plot A percent of set and recall data</b>: </label>
                <button id="download-data-csv-a">Download CSV</button>
            </div>
            <div class="mb-10" id="data-a"></div>
        </div>
        <div class="mb-10">
            <div class="mb-10">
            <label for="download-data-csv-b"><b>Plot B percent of set and recall data</b>: </label>
                <button id="download-data-csv-b">Download CSV</button>
            </div>
            <div class="mb-10" id="data-b"></div>
        </div>
            <div class="mb-10">
            <label for="download"><b>Predicted PS IDRs</b> (from input file, those w/ <i>N</i> â‰¥50): </label>
                <button id="download">Download FASTA</button>
            </div>
    </main>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/pako/2.0.4/pako.min.js"
        integrity="sha512-EAKzSKex+PXC0U9OG13r1059ysjrjkJEpZoONCnZa0mBROY28iBOOxZSErUVw1LzLr2+U5PhR7zPCPKidUVJqg=="
        crossorigin="anonymous" referrerpolicy="no-referrer"></script>
    <script src="https://cdn.plot.ly/plotly-2.4.2.min.js"></script>
    <script type="text/javascript" src="https://unpkg.com/tabulator-tables@5.4.3/dist/js/tabulator.min.js"></script>
    <script src="./js/part-a.js"></script>
    <script src="./js/part-b.js"></script>
    <script src="./js/constants.js"></script>
    <script src="./js/parse.js"></script>
    <script src="./js/app.js"></script>
</body>

</html>
